test/test_autograd.py

```diff
@@ -8460,6 +8460,93 @@ def _test_op(fn, inp, args):
         _test_op(torch.view_as_complex, torch.rand(2, 2), ())
         _test_op(torch.view_as_real, torch.rand(2, 2, dtype=torch.cfloat), ())
 
+    def test_forward_has_default_parameters(self):
+        class MyFunction(Function):
+            @staticmethod
+            def forward(x, y=3):
+                return torch.pow(x, y)
+
+            @staticmethod
+            def setup_context(ctx, inputs, output):
+                x, y = inputs
+                ctx.x = x
```

Comment:
nit: Saving directly on `ctx` is for non-tensors only. Tensors should be saved using the `ctx.save_for_backward(x)` API. This is still functionally correct for basic use cases (and for the purpose of this test), but (1) that tensor won't be freed during backward until the graph is cleared, and (2) other autograd features like activation checkpoint and saved tensor hooks won't correctly.